pkg/webui/console/api/index.js

```diff
@@ -76,24 +80,31 @@ export default {
       return instance.get(`${appRoot}/api/auth/token`)
     },
     async logout() {
-      let csrf = getCookieValue('_console_csrf')
-
-      if (!csrf) {
-        // If the csrf token has been deleted, we likely have some outside
-        // manipulation of the cookies. We can try to regain the cookie by
-        // making an AJAX request to the current location.
-        await axios.get(window.location)
-        csrf = getCookieValue('_console_csrf')
-
-        if (!csrf) {
-          // If we still could not retrieve the cookie, throw an error.
-          throw new Error('Could not retrieve the csrf token')
+      const headers = token => ({
+        headers: { 'X-CSRF-Token': token },
+      })
+      try {
+        return await axios.post(`${appRoot}/api/auth/logout`, undefined, headers(csrf))
+      } catch (error) {
+        if (
+          error.response &&
+          error.response.status === 403 &&
+          typeof error.response.data === 'string' &&
+          error.response.data.includes('CSRF')
+        ) {
+          // If the CSRF token is invalid, it likely means that the CSRF cookie
+          // has been deleted or became outdated. Making a new request to the
+          // current path can then retrieve a fresh CSRF cookie, with which
+          // the logout can be retried.
+          const csrfResult = await axios.get(window.location)
+          const freshCsrf = csrfResult.headers['x-csrf-token']
+          if (freshCsrf) {
+            return axios.post(`${appRoot}/api/auth/logout`, undefined, headers(freshCsrf))
+          }
         }
-      }
 
-      return instance.post(`${appRoot}/api/auth/logout`, undefined, {
-        headers: { 'X-CSRF-Token': csrf },
-      })
+        return error
```

Comment:
should this `throw` ? otherwise the user middleware processes this as a valid response and tries to extract `op_logout_uri`

Reply:
seems that it should not, but could you explain why?

Reply:
Sorry, I had a commit fixing this, but seems like it got missing somehow. Fixed now.